File: Task/royalscouts_frontend/lib/app/core/models/task.dart
class Task {
  final num id;
  final String title;
  final String status;
  final String origin;
  final String opc;
  final String npc;
  final String description;
  final String programmeType;
  final String spice;
  final int durationDays;
  final int durationHours;
  final int durationMins;
  final String participation;
  final String location;
  final int bonus;
  final int score;
  final String supervision;
  final String difficulty;
  final String createdBy;
  final String approvedBy;
  final String thingsNeeded;

  Task({
    required this.id,
    required this.title,
    required this.status,
    required this.origin,
    required this.opc,
    required this.npc,
    required this.description,
    required this.programmeType,
    required this.spice,
    required this.durationDays,
    required this.durationHours,
    required this.durationMins,
    required this.participation,
    required this.location,
    required this.bonus,
    required this.score,
    required this.supervision,
    required this.difficulty,
    required this.createdBy,
    required this.approvedBy,
    required this.thingsNeeded,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      status: json['status'],
      origin: json['origin'],
      opc: json['opc'],
      npc: json['npc'],
      description: json['description'],
      programmeType: json['programmeType'],
      spice: json['spice'],
      durationDays: json['durationDays'],
      durationHours: json['durationHours'],
      durationMins: json['durationMins'],
      participation: json['participation'],
      location: json['location'],
      bonus: json['bonus'],
      score: json['score'],
      supervision: json['supervision'],
      difficulty: json['difficulty'],
      createdBy: json['createdBy'],
      approvedBy: json['approvedBy'] ?? '',
      thingsNeeded: json['thingsNeeded'] ?? '',
    );
  }

  Map<String, dynamic> get values {
    return {
      'id': id,
      'title': title,
      'status': status,
      'origin': origin,
      'opc': opc,
      'npc': npc,
      'description': description,
      'programmeType': programmeType,
      'spice': spice,
      'durationDays': durationDays,
      'durationHours': durationHours,
      'durationMins': durationMins,
      'participation': participation,
      'location': location,
      'bonus': bonus,
      'score': score,
      'supervision': supervision,
      'difficulty': difficulty,
      'createdBy': createdBy,
      'approvedBy': approvedBy,
      'thingsNeeded': thingsNeeded,
    };
  }

  static Task setEmptyTask() {
    return Task(
      id: 0,
      title: '',
      origin: '',
      opc: '',
      npc: '',
      status: '',
      description: '',
      programmeType: '',
      spice: '',
      durationDays: 0,
      durationHours: 0,
      durationMins: 0,
      participation: '',
      location: '',
      bonus: 0,
      score: 0,
      supervision: '',
      difficulty: '',
      createdBy: '',
      approvedBy: '',
      thingsNeeded: '',
    );
  }
}

File: Task/royalscouts_backend/src/main/java/com/project/royalscouts/models/Task.java
package com.project.royalscouts.models;

import javax.persistence.*;

@Entity
@Table(name = "tasks")
public class Task extends AuditModel {

  @Column(name = "title")
  private String title;

  @Column(name = "status")
  private String status;

  @Column(name = "origin")
  private String origin;

  @Column(name = "opc")
  private String opc;

  @Column(name = "npc")
  private String npc;

  @Column(name = "description")
  private String description;

  @Column(name = "programmeType")
  private String programmeType;

  @Column(name = "spice")
  private String spice;

  @Column(name = "durationDays")
  private int durationDays;

  @Column(name = "durationHours")
  private int durationHours;

  @Column(name = "durationMins")
  private int durationMins;

  @Column(name = "participation")
  private String participation;

  @Column(name = "location")
  private String location;

  @Column(name = "bonus")
  private int bonus;

  @Column(name = "score")
  private int score;

  @Column(name = "supervision")
  private String supervision;

  @Column(name = "difficulty")
  private String difficulty;

  @Column(name = "approved_by")
  private String approvedBy;

  @Column(name = "created_by")
  private String createdBy;

  @Column(name = "things_needed")
  private String thingsNeeded;

  public Task() {
  }

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public String getStatus() {
    return status;
  }

  public void setStatus(String status) {
    this.status = status;
  }

  public String getOrigin() {
    return origin;
  }

  public void setOrigin(String origin) {
    this.origin = origin;
  }

  public String getOpc() {
    return opc;
  }

  public void setOpc(String opc) {
    this.opc = opc;
  }

  public String getNpc() {
    return npc;
  }

  public void setNpc(String npc) {
    this.npc = npc;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getProgrammeType() {
    return programmeType;
  }

  public void setProgrammeType(String programmeType) {
    this.programmeType = programmeType;
  }

  public String getSpice() {
    return spice;
  }

  public void setSpice(String spice) {
    this.spice = spice;
  }

  public int getDurationDays() {
    return durationDays;
  }

  public void setDurationDays(int durationDays) {
    this.durationDays = durationDays;
  }

  public int getDurationHours() {
    return durationHours;
  }

  public void setDurationHours(int durationHours) {
    this.durationHours = durationHours;
  }

  public int getDurationMins() {
    return durationMins;
  }

  public void setDurationMins(int durationMins) {
    this.durationMins = durationMins;
  }

  public String getParticipation() {
    return participation;
  }

  public void setParticipation(String participation) {
    this.participation = participation;
  }

  public String getLocation() {
    return location;
  }

  public void setLocation(String location) {
    this.location = location;
  }

  public int getBonus() {
    return bonus;
  }

  public void setBonus(int bonus) {
    this.bonus = bonus;
  }

  public int getScore() {
    return score;
  }

  public void setScore(int score) {
    this.score = score;
  }

  public String getSupervision() {
    return supervision;
  }

  public void setSupervision(String supervision) {
    this.supervision = supervision;
  }

  public String getDifficulty() {
    return difficulty;
  }

  public void setDifficulty(String difficulty) {
    this.difficulty = difficulty;
  }

  public String getApprovedBy() {
    return approvedBy;
  }

  public void setApprovedBy(String approvedBy) {
    this.approvedBy = approvedBy;
  }

  public String getCreatedBy() {
    return createdBy;
  }

  public void setCreatedBy(String createdBy) {
    this.createdBy = createdBy;
  }

  public String getThingsNeeded() {
    return thingsNeeded;
  }

  public void setThingsNeeded(String thingsNeeded) {
    this.thingsNeeded = thingsNeeded;
  }

  
}
